appveyor.yml

```diff
@@ -37,19 +43,23 @@ environment:
       PYTHON_HOME: C:\Python36-x64
       PYTHON_VERSION: 3.6
       PYTHON_ARCH: 64
-
 init:
-  - ps: echo $env:TOXENV
-  - ps: echo "Python $env:PYTHON_VERSION (${env:PYTHON_ARCH}bit) from $env:PYTHON_HOME"
-  - ps: ls C:\Python*
+- ps: echo $env:TOXENV
+- ps: echo "Python $env:PYTHON_VERSION (${env:PYTHON_ARCH}bit) from $env:PYTHON_HOME"
+- ps: ls C:\Python*
 install:
-  - git submodule update --init --recursive
-  - '%PYTHON_HOME%\Scripts\pip install --upgrade tox setuptools virtualenv wheel'
-  - '%PYTHON_HOME%\Scripts\virtualenv --version'
-  - '%PYTHON_HOME%\Scripts\pip --version'
-  - '%PYTHON_HOME%\Scripts\tox --version'
+- ps: $env:DO_DEPLOY="no"
```

Comment:
Yes there is. Whether `before_deploy` is reached, depends on if `deploy` will be reached, not vice versa. Before AppVeyor reaches `before_deploy` or `deploy`, it determines if the conditions are met to `deploy`. If the conditions are met to `deploy`, then it runs `before_deploy`, followed by `deploy`. So, this can be in any of the earlier stages but not in `before_deploy`.